gateway/api/context.py

```diff
@@ -0,0 +1,53 @@
+"""User context management using ContextVars."""
```

Comment:
Ok, this looks pretty cool. I'm not very familiar with `ContextVars`. Let me Monday to look at this more in-depth to try to understand it better, @avilches (today I'm totally done 😂).

cc/ @ElePT , @korgan00 just in case you want also take a look into this PR

Reply:
ContextVars is the replacement for threading local, where you can put arbitrary data in the context and read later without messing around sending the data in a parameter. The improvement with ContextVars is they support async/await (we don't use async/await in the gateway, but just in case in the future we start to use it)
Anyway, threading local or ContextVar are ideal to store info that could be needed in signals later.